src/libpanic_unwind/lib.rs

```diff
@@ -41,21 +41,33 @@ cfg_if::cfg_if! {
     if #[cfg(target_os = "emscripten")] {
         #[path = "emcc.rs"]
         mod real_imp;
-    } else if #[cfg(target_arch = "wasm32")] {
-        #[path = "dummy.rs"]
-        mod real_imp;
     } else if #[cfg(target_os = "hermit")] {
         #[path = "hermit.rs"]
         mod real_imp;
     } else if #[cfg(target_env = "msvc")] {
         #[path = "seh.rs"]
         mod real_imp;
-    } else {
+    } else if #[cfg(any(
+        all(target_family = "windows", target_env = "gnu"),
+        target_os = "cloudabi",
+        target_family = "unix",
```

Comment:
`target_family = "unix"` FWIW I think is the same as `unix`, right?

Reply:
Yes, it is the same.  I have been waffling on which style I like better, and I think I ended up being really inconsistent. I think part of the issue is that I've run into a number of users being confused what `cfg(unix)` means, which makes me feel it is better to be explicit. But `unix` and `windows` is conveniently short.  

One thing that surprises me is that it is possible to have `family="unix"` and `os="none"` at the same time.

Reply:
Eh either way is fine by me, I figured `unix` is a bit more idiomatic though. For `unix, target_os = "none"` being true that seems like it's either a mistake or something that hasn't been reviewed since it was added

Reply:
`x86_64-linux-kernel` is the only one.  I'm not sure what values would make sense for that, since it is unusual.

Reply:
Does the Linux kernel even have an UNIX API *inside* of itself? That doesn't sound right but maybe I just haven't seen it before.

Reply:
Filed #74247 for further discussion.

Reply:
Not sure if it was related or coincidental but I just saw #74257 being open.